### PR TITLE
Cluster and node names are required.

### PR DESCRIPTION
**Pull request by platon**

### CODE REVIEW

**Comment by platon** [comment]
Example of usage: 
riak-mesos node remove riak-default-1 -f
riak-mesos node remove --force riak-default-1

**Comment by platon** [comment]
Example of usage: 
riak-mesos node aae-status riak-default-1

**Comment by platon** [comment]
Example of usage: 
riak-mesos node status riak-default-1

**Comment by platon** [comment]
Example of usage: 
riak-mesos node ringready riak-default-1

**Comment by platon** [comment]
Example of usage:
riak-mesos node ringready riak-default-1

**Comment by platon** [comment]
Example of usage: `cluster create default riak-kv-2.1`

File: riak_mesos/config.py [hunk 1]
```diff
@@ -27,6 +27,25 @@ def __init__(self, config_file):
...
+        for key in keys:
+            value = value[key]
+        return value
+
+    def _get_resource_url(self, key):
+        return self._get_config_value('resources', key)
+
+    def _get_resource_fetch_urls(self):
+        fetch_urls = []
+        resources = self._get_config_value('resources')
```
Comment (sanmiguel) [inline riak_mesos/config.py:41]: Looks like this doesn't work in DCOS:

Comment (sanmiguel) [reply riak_mesos/config.py:41]:
```
 dcos riak config marathon
Traceback (most recent call last):
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/bin/dcos-riak", line 11, in <module>
    load_entry_point('dcos-riak-ts==0.0.1', 'console_scripts', 'dcos-riak')()
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/commands/cmd_config.py", line 41, in marathon
    click.echo(ctx.config.framework_marathon_string())
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/config.py", line 205, in framework_marathon_string
    return json.dumps(self.framework_marathon_json())
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/config.py", line 134, in framework_marathon_json
    mj['fetch'] = self._get_resource_fetch_urls()
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/config.py", line 41, in _get_resource_fetch_urls
    resources = self._get_config_value('resources')
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/config.py", line 33, in _get_config_value
    value = value[key]
KeyError: 'resources'
```

Comment (sanmiguel) [reply riak_mesos/config.py:41]: `dcos riak config riak-versions` does work, though...

File: riak_mesos/config.py [hunk 2]
```diff
@@ -64,7 +90,6 @@ def from_marathon(self, ctx):
         self._from_conf('auth-principal', None, 'RIAK_MESOS_PRINCIPAL', conf)
         self._from_conf('auth-secret-file', None,
                         'RIAK_MESOS_SECRET_FILE', conf)
-        self._from_conf('director', 'url', 'RIAK_MESOS_DIRECTOR_URL', conf)
         if 'RIAK_MESOS_DIRECTOR_CPUS' in conf:
             self._config['riak']['director']['cpus'] = \
                 float(conf['RIAK_MESOS_DIRECTOR_CPUS'])
```
Comment (sanmiguel) [inline riak_mesos/config.py:88]: Another missing config key here:

```
dcos riak director install  kv
Traceback (most recent call last):
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/bin/dcos-riak", line 11, in <module>
    load_entry_point('dcos-riak-ts==0.0.1', 'console_scripts', 'dcos-riak')()
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/commands/cmd_director.py", line 82, in install
    ctx.config.from_marathon(ctx)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/config.py", line 95, in from_marathon
    float(conf['RIAK_MESOS_DIRECTOR_CPUS'])
KeyError: 'director'
```

File: riak_mesos/commands/cmd_node.py [hunk 3]
```diff
@@ -39,19 +40,8 @@ def wait_for_service(ctx, **kwargs):
     wait_for_node(ctx, ctx.node)
```
Comment (sanmiguel) [inline riak_mesos/commands/cmd_node.py:40]: Another stack trace:

```
 dcos riak node wait-for-service  riak-kv-1
Traceback (most recent call last):
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/bin/dcos-riak", line 11, in <module>
    load_entry_point('dcos-riak-ts==0.0.1', 'console_scripts', 'dcos-riak')()
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/commands/cmd_node.py", line 40, in wait_for_service
    wait_for_node(ctx, ctx.node)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/util.py", line 31, in wait_for_node
    node_data = node_info(ctx, node)
  File "/Users/sanmiguel/.dcos/subcommands/riak/env/lib/python2.7/site-packages/riak_mesos/util.py", line 49, in node_info
    mesos_dns_cluster = fw + '-' + cluster + '.' + fw + '.mesos'
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

**Review by sanmiguel — CHANGES_REQUESTED** [review]
OK, I've found a couple of places there emit stacktraces and don't work - I've added the stacktraces I hit to the relevant lines in the tools. Let's try to smooth those out before we merge this

**Review by sanmiguel — APPROVED** [review]

**Pull request merged**